app/controllers/admin/domain_blocks_controller.rb

```diff
@@ -24,7 +24,7 @@ def batch
       flash[:alert] = I18n.t('admin.domain_blocks.no_domain_block_selected')
     rescue Mastodon::NotPermittedError
       flash[:alert] = I18n.t('admin.domain_blocks.not_permitted')
-    else
+    ensure
```

Comment:
Wait, this changes the flow here, doesn't it? By redirecting in to `admin_instances_path(limited: '1')` in the above cases when it previously did not.

Reply:
Yes it does -- I tried to note that in description, but could have added more context ...

Currently when the `ParameterMissing` path is hit, the flash gets set and the response searches for a view template for "batch". There is not one, so it returns an empty 204 no content response. I interpreted this as a bug (?), since this same pattern is used in other controllers (look at admin/custom_emojis or admin/email_domain_blocks for example) but not here.

If for whatever reason all the rest of them use ensure but this should not, I can undo that portion of this.

Reply:
Given that the redirect has `notice: I18n.t('admin.domain_blocks.created_msg')`, error cases would leat to a successful creation message be erroneously displayed.

That being said it does seem like the rescues are missing a redirect.

Maybe something like the following? (do not apply it as-is as it would duplicate the redirect line)
```suggestion
    else
      flash[:notice] = I18n.t('admin.domain_blocks.created_msg')
    ensure
      redirect_to admin_instances_path(limited: '1')
```